src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1377,6 +1377,33 @@ class LedgerMasterImp
         return mCompleteLedgers.toString ();
     }
 
+    boost::optional <uint32_t>
+    getCloseTimeBySeq (LedgerIndex ledgerIndex) override
+    {
+        uint256 hash = getHashBySeq (ledgerIndex);
+        return hash.isNonZero() ? getCloseTimeByHash (hash) : boost::none;
+    }
+
+    boost::optional <uint32_t>
+    getCloseTimeByHash (LedgerHash const& ledgerHash) override
+    {
+        auto node = app_.getNodeStore().fetch (ledgerHash);
+        if (node &&
+            (node->getData().size() >= 120))
+        {
+            SerialIter it (node->getData().data(), node->getData().size());
+            if (it.get32() == HashPrefix::ledgerMaster)
+            {
+                it.skip (
+                    4+8+32+    // seq drops parentHash
+                    32+32+4);  // txHash acctHash parentClose
+                return it.get32();
```

Comment:
It took me a bit to get the 120, but I believe it is correct.  You need to add 4 for the prefix read before the skip, and 4 more for the hash that needs to be read after the skip.